pelicun/uq.py

```diff
@@ -1278,12 +1283,89 @@ def __init__(
         """
         if truncation_limits is None:
             truncation_limits = np.array((np.nan, np.nan))
+
+        # For backwards compatibility, cast to a numpy array if
+        # given a tuple or list.
+        if isinstance(theta, (list, tuple)):
+            theta = np.array(theta)
+        if isinstance(truncation_limits, (list, tuple)):
+            truncation_limits = np.array(truncation_limits)
+
+        # Verify type
+        if theta is not None:
+            assert isinstance(
+                theta, np.ndarray
+            ), 'Parameter `theta` should be a numpy array.'
+            assert theta.ndim in {
+                1,
+                2,
+            }, 'Parameter `theta` can only be a 1D or 2D array.'
+            theta = np.atleast_1d(theta)
+
+        assert isinstance(
+            truncation_limits, np.ndarray
+        ), 'Parameter `truncation_limits` should be a numpy array.'
+        assert truncation_limits.ndim in {
+            1,
+            2,
+        }, 'Parameter `truncation_limits` can only be a 1D or 2D array.'
+        # 1D corresponds to fixed parameters.
+        # 2D corresponds to variable parameters (different in each
+        # realization).
+
+        self.theta = theta
+        self.truncation_limits = truncation_limits
+
         super().__init__(
             name=name,
             f_map=f_map,
             anchor=anchor,
         )
 
+    def constant_parameters(self) -> bool:
+        """
+        If the RV has constant or variable parameters.
+
+        Constant parameters are the same in each realization.
+
+        Returns
+        -------
+        bool
+          True if the parameters are fixed, false otherwise.
+
+        """
+        assert self.theta is not None
```

Comment:
I found it, it's the EmpiricalRandomVariable. I wonder how important it is for that variable to have a `theta=None` assignment. From a pragmatic point of view, couldn't we just use `theta=np.array([np.nan, np.nan])`?

Reply:
I agree with using `theta` directly instead of `_raw_sample` in `EmpiricalRandomVariable`. We talked about this and this update is coming in a future PR.